module-4/backend/week_two.md

```diff
@@ -1,22 +1,39 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's one difference between ES5 and ES6?
-2. What's the difference between asynchronous and synchronous JavaScript? 
++ ES5 works on IE. ES6 does not have the same browser support. ES6 lets you do string interpolation like Ruby and you can build classes more easily.
+2. What's the difference between asynchronous and synchronous JavaScript?
++ Asynchronous functions work out of line with the other functions of the program, while synchronous functions happen in line.
 3. What are the four pillars of Object Oriented programming?
++ Encapsulation
++ Abstraction
++ Inheritance
++ Polymorphism
 4. What are some tools available in JavaScript to help you write object oriented code?
++ I'm not sure on the tools, but I find getting comfortable with a debugger allows me to focus on design more.
 5. What are some key concepts to be aware of when refactoring your JavaScript?
++ I have found the biggest thing I am missing in my first attempt at a problem in JS is to not use classes and follow OOP.
 6. What's a callback function and what are some reasons when we use/need callback functions?
++ A callback function is when you tag a function onto another and use that function as a parameter. Because functions are first order objects, we can treat them as strings or arrays.
 7. What's the scope of variables in Javascript?
++ They are scoped to within the function that are created in, unless they are made outside of a function and become global variables.
 8. What's the difference between `==` and `===` in JavaScript?
++ `==` accounts for differences in type and `===` does not.
 9. Why do front end frameworks exist?
++ For the same reasons that back end frameworks exist, to make creating easier and more efficient.
 
 #### Review  
 
 10. Why do people say "HTTP is stateless"?
++ Because the server doesn't remember anything about the communication it does with different partners.
 11. Describe a RESTful API.
++ Application that has GET, PUT, POST and DELETE
```

Comment:
not entirely. the application uses those http verbs and matches those with the intended CRUD functionality in addition to the url path